src/Stratis.Bitcoin.IntegrationTests/SegWitTests.cs

```diff
@@ -30,44 +28,68 @@ public void TestSegwitActivation()
                     fullNodeBuilder
                         .UseConsensus()
                         .UseBlockStore()
+                        .UseWallet()
                         .UseMempool()
-                        .UseBlockNotification()
-                        .UseTransactionNotification()
                         .AddMining()
-                        .UseWallet()                        
-                        .UseApi()
                         .AddRPC();
                 });
 
-                WalletManager wm1 = node1.FullNode.NodeService<IWalletManager>() as WalletManager;
-                wm1.CreateWallet("Test1", "alice1");
-
-                RPCClient rpc1 = node1.CreateRPCClient();
+                RPCClient stratisNodeRpc = stratisNode.CreateRPCClient();
                 RPCClient coreRpc = coreNode.CreateRPCClient();
 
-                coreRpc.AddNode(node1.Endpoint, false);
-                rpc1.AddNode(coreNode.Endpoint, false);
+                coreRpc.AddNode(stratisNode.Endpoint, false);
+                stratisNodeRpc.AddNode(coreNode.Endpoint, false);
 
-                coreRpc.Generate(450);
+                // core (in version 0.15.1) only mines segwit blocks above a certain height on regtest
+                // future versions of core will change that behaviour so this test may need ot be changed in the future
+                // see issue for more details https://github.com/stratisproject/StratisBitcoinFullNode/issues/1028
+                var prevSegwitDeployment = Network.RegTest.Consensus.BIP9Deployments[BIP9Deployments.Segwit];
+                Network.RegTest.Consensus.BIP9Deployments[BIP9Deployments.Segwit] = new BIP9DeploymentsParameters(1, 0, DateTime.Now.AddDays(50).ToUnixTimestamp());
 
-                BIP9DeploymentsArray bip9Constants = node1.FullNode.Network.Consensus.BIP9Deployments;
-                IConsensusLoop consensusLoop = node1.FullNode.NodeService<IConsensusLoop>();
-                ThresholdState[] bip9State = consensusLoop.NodeDeployments.BIP9.GetStates(node1.FullNode.Chain.Tip.Previous);
+                try
+                {
+                    // generate 450 blocks, block 431 will have segwit activated.
+                    coreRpc.Generate(450);
 
-                Money amount = new Money(5.0m, MoneyUnit.BTC);
-                HdAddress destination1 = wm1.GetUnusedAddress(new WalletAccountReference("alice1", "account 0"));
+                    TestHelper.WaitLoop(() => stratisNode.CreateRPCClient().GetBestBlockHash() == coreNode.CreateRPCClient().GetBestBlockHash());
 
-                coreRpc.SendToAddress(BitcoinAddress.Create(destination1.Address, Network.RegTest), amount);
+                    // segwit activation on Bitcoin regtest
+                    // - On regtest deployment state changes every 144 block, the threshold for activating a rule is 108 blocks.
+                    // segwit deployment status should be:
+                    // - Defined up to block 142
```

Comment:
Dots for those 4 paragraphs (or remove dot for the 1 that is above) for consistency.